runtime-sdk-macros/src/lib.rs

```diff
@@ -0,0 +1,32 @@
+#![feature(proc_macro_diagnostic)]
+#![deny(rust_2018_idioms)]
+
+use proc_macro::TokenStream;
+
+mod error_derive;
+mod event_derive;
+mod generators;
+#[cfg(test)]
+mod test_utils;
+mod version_from_cargo;
+
+/// Derives the `Event` trait on an enum.
+#[proc_macro_derive(Event, attributes(sdk))]
+pub fn event_derive(input: TokenStream) -> TokenStream {
+    let input = syn::parse_macro_input!(input as syn::DeriveInput);
+    event_derive::derive_event(input).into()
+}
+
+/// Derives the `Error` trait on an enum.
+// The helper attribute is `runtime_error` to avoid conflict with `thiserror::Error`.
```

Comment:
Hm?

Reply:
🤦 but in my defense, I was waiting for feedback on the final helper attribute name

Reply:
He he, so as I mentioned above I think it could be `sdk_error` and `sdk_event`, what do you think?